domain/application/state/resource.go

```diff
@@ -69,13 +386,103 @@ func (st *State) OpenUnitResource(
 	return resource.Resource{}, nil
 }
 
-// SetRepositoryResources sets the "polled" resource
-// s for the
+// SetRepositoryResources sets the "polled" resources for the
 // application to the provided values. The current data for this
 // application/resource combination will be overwritten.
+// Returns [apperrors.ApplicationNotFound] if the application id doesn't belong to a valid application.
 func (st *State) SetRepositoryResources(
```

Comment:
I assume it is :)  

I followed the description in documentation. I think it will be used combined with other state method. Maybe it will be refactored a bit during wiring.

Reply:
Fair enough, we'll cross this bridge when we get there.